macros/calibrateThrDac.py

```diff
@@ -112,16 +113,26 @@
     from gempython.gemplotting.utils.threshAlgos import calibrateThrDAC
     from gempython.utils.gemlogger import printGreen, printRed
     from gempython.utils.wrappers import runCommand
+    from gempython.gemplotting.utils.namespace import Namespace
     import os, sys, traceback
     try:
-        retCode = calibrateThrDAC(
-                inputFile=args.filename,
-                fitRange=args.fitRange,
-                listOfVFATs=args.listOfVFATs,
-                noLeg=args.noLeg,
-                outputDir=outputDir,
-                savePlots=args.savePlots,
-                debug=args.debug)
+
+        args = Namespace(
+            inputFile=args.filename,
+            fitRange=args.fitRange,
+            listOfVFATs=args.listOfVFATs,
+            maxEffPedPercent=args.maxEffPedPercent,
+            highNoiseCut=args.highNoiseCut,
+            deadChanCutLow=args.deadChanCutLow,
+            deadChanCutHigh=args.deadChanCutHigh,
+            noLeg=args.noLeg,
+            outputDir=outputDir,
+            savePlots=args.savePlots,
+            debug=args.debug
+        )
```

Comment:
Here it seems you're just reseting `args` to itself and changing the name of one member and adding another (e.g. `inputFile` and `outputDir`).  Rather than reseting the whole object it would be simpler to change line 88 to:

```python
parser.add_argument("filename", type=str, dest="inputFile",
```

Then the `filename` argument is stored in `args.inputFile` and then to just add `outputDir` as a single line, i.e.:

```python
args.outputDir = outputDir
``` 

This is much simpler than what you've done in lines 120-132.

Finally typically you should place the minimum amount of code inside of a `try` block and have what is inside the `try` block be only relevant to your exception handling.  Right now the exception handling is designed to catch only `IOError`s and manipulating `args` has nothing to do with this.  So the only thing that should be in this `try` block should be line 134.

Reply:
Instead of adding the destination for the `filename` argument, I just changed `filename` to `inputFile`. I think adding a different destination for a positional argument is not allowed.

Reply:
I would not change the interface that the user sees.  And the `dest` argument for `add_argument` can be used regardless of whether or not the the argument is positional or optional.

https://docs.python.org/2/library/argparse.html#the-add-argument-method